src/sentry/utils/performance_issues/detectors/http_overhead_detector.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import annotations
+
+import urllib.parse
+from dataclasses import dataclass
+from typing import Optional
+
+from sentry import features
+from sentry.issues.grouptype import PerformanceHTTPOverheadGroupType
+from sentry.issues.issue_occurrence import IssueEvidence
+from sentry.models import Organization, Project
+
+from ..base import (
+    DetectorType,
+    PerformanceDetector,
+    does_overlap_previous_span,
+    get_notification_attachment_body,
+    get_span_evidence_value,
+)
+from ..performance_problem import PerformanceProblem
+from ..types import Span
+
+
+@dataclass
+class ProblemIndicator:
+    """
+    Keep span data that will be used to store the problem together.
+    Has a monotonic queue depth to know if spans hit the parallel limit without walking all spans again.
+    """
+
+    span: Span
+    delay: float
+    queue_depth: int = 0
+
+
+class HTTPOverheadDetector(PerformanceDetector):
+    """
+    Detect HTTP/1.1 overhead in http spans outside of response time for that endpoint using
+    Browser reporting of PerformanceNavigationTiming. Parent-child span hierarchy
+    is ignored as there is an external limit (browser http/1.1 request limit) solely based on time.
+
+      [-------- transaction -----------]
+        [0] https://service.io/api/book/1      - request delay ~0ms
+        ----[1] https://service.io/api/book/2  - request delay ~0ms
+        ----[2] https://service.io/api/book/3  - request delay ~0ms
+        ...
+        ----[5] https://service.io/api/book/6  - request delay 400ms - Hit ~6 connection limit.
+        ----[6] https://service.io/api/book/7  - request delay 600ms - Over threshold, triggers detection.
+        ...
+    """
+
+    __slots__ = "stored_problems"
+
+    type = DetectorType.HTTP_OVERHEAD
+    settings_key = DetectorType.HTTP_OVERHEAD
+
+    def init(self):
+        self.stored_problems: dict[str, PerformanceProblem] = {}
+        self.location_to_indicators = {}
+
+    def visit_span(self, span: Span) -> None:
+        span_data = span.get("data", {})
+        if not self._is_span_eligible(span) or not span_data:
+            return
+
+        url = span_data.get("url", "")
+        span_start = span.get("start_timestamp", 0) * 1000
+        request_start = span_data.get("http.request.request_start", 0) * 1000
+
+        if not url or not span_start or not request_start:
+            return
+
+        if url.startswith("/"):
+            location = "/"
+        else:
+            parsed_url = urllib.parse.urlparse(url)
+            location = parsed_url.netloc
+
+        if not location:
+            return
+
+        if location not in self.location_to_indicators:
```

Comment:
Not a big deal, but you could've used a default dict to avoid having to set this manually

Reply:
Always forget it exists when switching from js